src/builtin.rs

```diff
@@ -27,6 +30,30 @@ pub fn exit(_args : &[&str]) -> io::Result<i32> {
     process::exit(0)
 }
 
+pub fn echo(args : &[&str]) -> io::Result<i32> {
+    let mut opts = Options::new();
+
+    opts.optflag("n", "", "Suppress new lines");
+
+    let matches = match opts.parse(&args[..]) {
```

Comment:
I'd prefer this to be a try.  Also you can change ```&args[..]``` to ```args``` since it's already a &[].